### PR TITLE
[PWGCF] support for EP angle and few more event cuts

### PR DESCRIPTION
**Pull request by abilandz**

### CODE REVIEW

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 1]
```diff
@@ -1366,6 +1394,10 @@ void DefaultCuts()
   ec.fdEventCuts[eImpactParameter][eMin] = lImpactParameter[eMin];
   ec.fdEventCuts[eImpactParameter][eMax] = lImpactParameter[eMax];
 
+  auto lEventPlaneAngle = (vector<float>)cf_ec.cfEventPlaneAngle;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1397]: Use instead `std::vector<float>` The `linter` will be happier

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 2]
```diff
@@ -5481,9 +5656,10 @@ void FillEventHistograms(T1 const& collision, T2 const& tracks, eBeforeAfter ba)
 
   // -----------------------------------------------------------------------------
 
-  // d) Fill only simulated(Run 3 specific):
+  // d) Fill only simulated (Run 3 specific):
   if constexpr (rs == eSim) {
     // !eh.fEventHistograms[eImpactParameter][eSim][ba] ? true : eh.fEventHistograms[eImpactParameter][eSim][ba]->Fill(collision.impactParameter()); // yes, because in this branch 'collision' is always aod::McCollision
+    // !eh.fEventHistograms[eEventPlaneAngle][eSim][ba] ? true : eh.fEventHistograms[eEventPlaneAngle][eSim][ba]->Fill(collision.eventPlaneAngle()); // yes, because in this branch 'collision' is always aod::McCollision
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:5662]: Remember not leaving code commented

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 3]
```diff
@@ -5515,6 +5691,7 @@ void FillEventHistograms(T1 const& collision, T2 const& tracks, eBeforeAfter ba)
   // f) Fill only simulated (Run 1 and 2 specific): // In case there is some corner case between Run 1 and Run 2, simply branch further this one
   if constexpr (rs == eSim_Run2 || rs == eSim_Run1) {
     // !eh.fEventHistograms[eImpactParameter][eSim][ba] ? true : eh.fEventHistograms[eImpactParameter][eSim][ba]->Fill(collision.impactParameter()); // yes, because in this branch 'collision' is always aod::McCollision
+    // !eh.fEventHistograms[eEventPlaneAngle][eSim][ba] ? true : eh.fEventHistograms[eEventPlaneAngle][eSim][ba]->Fill(collision.eventPlaneAngle()); // yes, because in this branch 'collision' is always aod::McCollision
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:5694]: Same here

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 4]
```diff
@@ -10428,6 +10605,8 @@ void DetermineReferenceMultiplicity(T const& collision)
       ebye.fReferenceMultiplicity = collision.multFT0M();
     } else if (ec.fsEventCuts[eReferenceMultiplicityEstimator].EqualTo("multNTracksPV", TString::kIgnoreCase)) {
       ebye.fReferenceMultiplicity = collision.multNTracksPV();
+    } else if (ec.fsEventCuts[eReferenceMultiplicityEstimator].EqualTo("multNTracksGlobal", TString::kIgnoreCase)) {
+      // ebye.fReferenceMultiplicity = collision.multNTracksGlobal(); // TBI 20241209 not validated yet
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:10609]: ...

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 5]
```diff
@@ -10438,6 +10617,7 @@ void DetermineReferenceMultiplicity(T const& collision)
       qa.fReferenceMultiplicity[eMultFT0C] = collision.multFT0C();
       qa.fReferenceMultiplicity[eMultFT0M] = collision.multFT0M();
       qa.fReferenceMultiplicity[eMultNTracksPV] = collision.multNTracksPV();
+      // qa.fReferenceMultiplicity[eMultNTracksGlobal] = collision.multNTracksGlobal(); // TBI 20241209 not validated yet
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:10620]: ...

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ab.cxx [hunk 6]
```diff
@@ -29,11 +29,12 @@ using BCs_Run3 = soa::Join<aod::BCs, aod::Timestamps, aod::Run3MatchedToBCSparse
...
 // Remark 2: For consistency with notation below, drop _Run3 and instead use _Run2 and _Run1
 
+// using EventSelection = soa::Join<aod::EvSels, aod::Mults, aod::MultsGlobal, aod::CentFT0Cs, aod::CentFT0Ms, aod::CentFV0As, aod::CentNTPVs>; // TBI 20241209 validating "MultsGlobal"
+//  for using collision.multNTracksGlobal()
 using EventSelection = soa::Join<aod::EvSels, aod::Mults, aod::CentFT0Cs, aod::CentFT0Ms, aod::CentFV0As, aod::CentNTPVs>;
 using CollisionRec = soa::Join<aod::Collisions, EventSelection>::iterator; // use in json "isMC": "true" for "event-selection-task"
 using CollisionRecSim = soa::Join<aod::Collisions, aod::McCollisionLabels, EventSelection>::iterator;
-// using CollisionRecSim = soa::Join<aod::Collisions, aod::McCollisionLabels, aod::MultsExtraMC, EventSelection>::iterator;
-// using CollisionRecSim = soa::Join<aod::Collisions, aod::McCollisionLabels, aod::MultsExtraMC, EventSelection>::iterator;
+// using CollisionRecSim = soa::Join<aod::Collisions, aod::McCollisionLabels, aod::MultsExtraMC, EventSelection>::iterator; // TBI 20241210 validating "MultsExtraMC" for multMCNParticlesEta08
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ab.cxx:37]: ...

File: PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ab.cxx [hunk 7]
```diff
@@ -29,11 +29,12 @@ using BCs_Run3 = soa::Join<aod::BCs, aod::Timestamps, aod::Run3MatchedToBCSparse
...
 using CollisionSim = aod::McCollision;
 using TracksRec = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::TrackSelection>;
 // using TrackRec = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::TrackSelection>::iterator;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ab.cxx:40]: ...

**Review by victor-gonzalez — APPROVED** [review]

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h (hunk 2 above)
Comment (abilandz) [reply PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:5662]: It's only temporarily commented out, I will use this code eventually, after I resolve and validate a few other thingies related to it.

**Pull request merged**